app/src/main/java/org/oppia/android/app/recyclerview/BindableAdapter.kt

```diff
@@ -99,7 +99,7 @@ class BindableAdapter<T : Any> internal constructor(
    * The base builder for [BindableAdapter]. This class should not be used directly--use either
    * [SingleTypeBuilder] or [MultiTypeBuilder] instead.
    */
-  abstract class BaseBuilder(private val fragment: Fragment) {
+  abstract class BaseBuilder internal constructor(fragment: Fragment) {
```

Comment:
Given that we no longer need to expose this class outside of the adapter, can we make both the class & its constructor private?

Reply:
@BenHenning changing it to private shows error "Cannot access '<init>': it is private in 'BaseBuilder'" please advice on this.

Reply:
Per our meeting, I think I suggested something along the lines of hiding the inline factory function behind a non-inline variant, but I actually don't think that's possible since we're using a reified type. This unfortunately has to be public I think.